pkg/monitoring/rules/recordingrules/virt.go

```diff
@@ -73,7 +73,7 @@ func virtRecordingRules(namespace string) []operatorrules.RecordingRule {
 			},
 			MetricType: operatormetrics.GaugeType,
 			Expr: intstr.FromString(
-				fmt.Sprintf("sum(kubevirt_virt_operator_ready_status{namespace='%s'}) or vector(0)", namespace),
+				fmt.Sprintf("sum(kube_pod_status_ready{pod=~'virt-operator-.*', condition='true', namespace='%s'}) or vector(0)", namespace),
```

Comment:
this is not quite the same since the readiness probe is configured for the metrics endpoint

```
    readinessProbe:
      failureThreshold: 3
      httpGet:
        path: /metrics
        port: 8443
        scheme: HTTPS
      initialDelaySeconds: 5
      ...
```

which would mean the pod would be ready right after https://github.com/kubevirt/kubevirt/pull/12071/files#diff-efc262b621a95dbf2a12cacb2b5b8c7ae391f0bd42af1b84892fa3855be0b465R337


Maybe makes sense to propose to change the readiness probe to the webhook server endpoint https://github.com/kubevirt/kubevirt/pull/12071/files#diff-efc262b621a95dbf2a12cacb2b5b8c7ae391f0bd42af1b84892fa3855be0b465R398 and move the setup of the leader election (https://github.com/kubevirt/kubevirt/pull/12071/files#diff-efc262b621a95dbf2a12cacb2b5b8c7ae391f0bd42af1b84892fa3855be0b465R404) to be done before starting listening on that endpoint

Reply:
How is the fact that the readiness probe is configured for the metrics endpoint is related to this PR? 
IIUC, the readiness probe affects the way the pod is considered ready, and this is an important discussion, but unrelated to the changes in this PR IMO.
The alert would still need to use `kube_pod_status_ready`, wouldn't it?